iOS/Flows/Conversation/Cells/ConversationMessageCell.swift

```diff
@@ -244,19 +192,20 @@ extension ConversationMessageCell: UICollectionViewDataSource, UICollectionViewD
                         layout collectionViewLayout: UICollectionViewLayout,
                         sizeForItemAt indexPath: IndexPath) -> CGSize {
 
-        let width = collectionView.width
+        var width = collectionView.width
         var height: CGFloat = 50
 
         // The heights of all cells in a section are the same as the front most cell in that section.
-        switch indexPath.section {
-        case 0:
-            guard let latestMessage = self.otherMessages.last else { break }
-            height = MessageSubcell.getHeight(withWidth: width, message: latestMessage)
-        case 1:
-            guard let latestMessage = self.currentUserMessages.last else { break }
+        if let sectionID = ConversationMessageSection(rawValue: indexPath.section),
+           let lastItem = self.dataSource.itemIdentifiers(in: sectionID).last,
+            let latestMessage = ChatClient.shared.messageController(cid: lastItem.channelID,
+                                                                    messageId: lastItem.messageID).message {
+
             height = MessageSubcell.getHeight(withWidth: width, message: latestMessage)
-        default:
-            break
+
+            // Shrink down cells as they get closer to the bottom of the stack.
+            let stackIndex = self.dataSource.getStackIndex(forIndexPath: indexPath)
+            width -= CGFloat(stackIndex) * 15
```

Comment:
I made convenience function for getting the stack index. Index 0 is the front most in the stack.